lib/bootstrap_form/helpers/bootstrap.rb

```diff
@@ -48,15 +48,21 @@ def static_control(*args, &block)
         options = args.extract_options!
         name = args.first
 
-        html = if block_given?
+        value = if block_given?
           capture(&block)
+        elsif options[:value]
+          options[:value]
         else
           object.send(name)
         end
 
-        form_group_builder(name, options) do
-          content_tag(:p, html, { class: static_class }.merge(options[:id].present? ? { id: options[:id] } : {}))
-        end
+        static_options = {
+          value: value,
+          readonly: true,
+          control_class: static_class
```

Comment:
If the user passed in an `option[:control_class]`, we want to preserve it and add `static_class` to the list of `control_class`es. I believe the way this is merged below will overwrite whatever the user passes in for `option[:control_class]`

Reply:
Can do, is `control_class` specifically put on the `input`(or whatever form element) and the `class` option put on the surrounding div? I'm not entirely certain what the difference between these two options is.

Reply:
According to the documentation, the `:control_class` string is added to the classes that are put on the `input` tag. The `:wrapper_class` string is added to the classes that are put on the `div.form-group` wrapper. `:class`.

It's very hard to tell what exactly is going on in that code. Even though I've spent the last two months off and on staring at that code, I'm going to have to take some time be sure. Sorry I can't be more help right now. I'll try to take a look at this tonight (Pacific Time).

It might be worth your time to find some test cases that test those options, and see what they do. Maybe add a few test cases if you're still in doubt about what the code does.

Reply:
OK. My bad. Sorry if I misled you. `:control_class` will not apply the `.form-control` that we need. However, if you set `options[:class] = static_class` it works (as I think you figured out a day or two ago). My apologies again for giving you the run-around, but if you can put it back to using `control` instead of `control_class`, we can merge this PR and move on. Thanks in advance.

Reply:
BTW: I figured this out by adding a test in a local branch:
```
  test "control_class passed correctly" do
    expected = <<-HTML.strip_heredoc
      <div class="form-group">
        <label class="required" for="user_email">Email</label>
        <input class="my-class" id="user_email" name="user[email]" type="text" value="steve@example.com" />
      </div>
    HTML
    assert_equivalent_xml expected, @builder.text_field(:email, control_class: "my-class")
  end
```